addons/controllers/clusterbootstrap_controller.go

```diff
@@ -77,33 +77,31 @@ type ClusterBootstrapConfig struct {
 	CNISelectionClusterVariableName string
 }
 
+// ClusterBootstrapWatchInputs contains the inputs for Watches set in ClusterBootstrap
+type ClusterBootstrapWatchInputs struct {
```

Comment:
I would use the existing type [WatchesInput](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/builder/controller.go#L116)

Reply:
The fields in the existing type [WatchesInput](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/builder/controller.go#L116) are private and cannot be accessed outside of that package. To create an array of that type, we would need to assign values to those private fields.